src/Console/ModelsCommand.php

```diff
@@ -477,21 +477,35 @@ protected function getPropertiesFromTable($model)
             if (!$column->getNotnull()) {
                 $this->nullableColumns[$name] = true;
             }
+
+            $propertyType = $this->getTypeInModel($model, $type);
+            $nullable = !$column->getNotnull();
+
             $this->setProperty(
                 $name,
-                $this->getTypeInModel($model, $type),
+                $propertyType,
                 true,
                 true,
                 $comment,
-                !$column->getNotnull()
+                $nullable
             );
+
             if ($this->write_model_magic_where) {
+                $truePropertyType = $this->getTruePropertyType(
+                    $propertyType,
+                    $nullable
+                );
+
                 $this->setMethod(
                     Str::camel('where_' . $name),
                     $this->getClassNameInDestinationFile($model, \Illuminate\Database\Eloquent\Builder::class)
                     . '|'
                     . $this->getClassNameInDestinationFile($model, get_class($model)),
-                    ['$value']
+                    [
+                        empty($truePropertyType)
+                            ? '$value'
+                            : "{$truePropertyType} \$value",
```

Comment:
No need for `{`/`}`
```suggestion
                            : "$truePropertyType \$value",
```

Reply:
It's habit for me; my personal preference. I'll change it, but if you don't want it, then maybe look into updating the sniffs.